core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -710,7 +710,11 @@ private[spark] class TaskSetManager(
       logInfo(s"Killing attempt ${attemptInfo.attemptNumber} for task ${attemptInfo.id} " +
         s"in stage ${taskSet.id} (TID ${attemptInfo.taskId}) on ${attemptInfo.host} " +
         s"as the attempt ${info.attemptNumber} succeeded on ${info.host}")
-      sched.backend.killTask(attemptInfo.taskId, attemptInfo.executorId, true)
+      sched.backend.killTask(
+        attemptInfo.taskId,
+        attemptInfo.executorId,
+        interruptThread = true,
+        reason = "another attempt succeeded")
```

Comment:
Ok let's leave this as-is -- seems too complicated to have a longer and shorter reason (and unlike the reason above, this one is per-task, so hard to summarize on the stage page)